src/main/java/seedu/address/model/person/Email.java

```diff
@@ -58,9 +62,19 @@ public String toString() {
 
     @Override
     public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Email // instanceof handles nulls
-                && value.equals(((Email) other).value)); // state check
+        if (other == this) {
+            return true;
+        } else if (other instanceof Email) {
+            if (value == null) {
+                return Objects.isNull(((Email) other).value);
+            } else if (Objects.isNull(((Email) other).value)) {
+                return false;
+            } else {
+                return value.equals(((Email) other).value);
+            }
+        } else {
+            return false;
+        }
```

Comment:
I've been seeing this a lot, I wonder if there's a neater way to do this in a way that handles nulls...